src/Workspaces/Core/Portable/Diagnostics/DiagnosticProvider.cs

```diff
@@ -1,60 +0,0 @@
-﻿// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MIT license.
-// See the LICENSE file in the project root for more information.
-
-#nullable disable
-
-using System;
-using Microsoft.CodeAnalysis.Diagnostics;
-using Microsoft.CodeAnalysis.SolutionCrawler;
-
-namespace Microsoft.CodeAnalysis
-{
-    /// <summary>
-    /// Provide a way for users to turn on and off analyzing workspace for compiler diagnostics
-    /// </summary>
-    internal static class DiagnosticProvider
-    {
-        public static void Enable(Workspace workspace, Options options)
-        {
-            var service = workspace.Services.GetService<ISolutionCrawlerRegistrationService>();
-
-            var newOptions = GetOptions(workspace, options);
-            workspace.TryApplyChanges(workspace.CurrentSolution.WithOptions(newOptions));
-            service.Register(workspace);
```

Comment:
>service.Register(workspace) [](http://example.com/codeflow?start=12&length=27)

service.Register should probably still be called

Reply:
done.